core/src/io/anuke/mindustry/io/JsonIO.java

```diff
@@ -57,6 +57,21 @@ public TeamData read(Json json, JsonValue jsonData, Class type){
                 return out;
             }
         });
+
```

Comment:
You should also do `setElementType(Rules.class, "loadout", ItemStack.class);` near the top to prevent the serializer from writing item classes unnecessarily.